File: LeetCode/cpp/1006.cpp
class Solution {
public:
    int clumsy(int N) {
        vector<int> nums;
        if (N >= 4) {
            nums.push_back(N * (N - 1) / (N - 2) + (N - 3));
            N -= 4;
        }
        while (N >= 4) {
            nums.push_back(N * (N - 1) / (N - 2) - (N - 3));
            N -= 4;
        }
        int temp = 0;
        if (N > 0) {
            temp = N;
            --N;
            if (N > 0) {
                temp *= N;
                --N;
                if (N > 0) {
                    temp /= N;
                    --N;
                }
            }
        }
        if (nums.empty())
            return temp;
        int res = nums[0];
        int i = 1;
        while (i < nums.size()) {
            res -= nums[i];
            ++i;
        }
        return res - temp;
    }
};

File: LeetCode/python3/1006.py
class Solution:
    def clumsy(self, N: int) -> int:
        if N <= 2:
            return N
        sum = N * (N - 1) // (N - 2)
        N -= 3
        while N > 0:
            sum += N
            temp = 0
            if N - 1 > 0:
                temp = N - 1
                if N - 2 > 0:
                    temp *= N - 2
                    if N - 3 > 0:
                        temp //= N - 3
            sum -= temp
            N -= 4
        return sum
